redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -0,0 +1,339 @@
+                    }
+                    return;
+                },
+                hwIsolationDbusName, resourceIsolatedHwEntry,
+                "xyz.openbmc_project.Object.Delete", "Delete");
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        resourceObjPath.str + "/isolated_hw_entry",
+        "org.freedesktop.DBus.Properties", "Get",
+        "xyz.openbmc_project.Association", "endpoints");
+}
+
+/**
+ * @brief API used to process hardware (aka resource) isolation request
+ *        This API can be used to any redfish resource if that redfish
+ *        supporting isolation feature (the resource can be isolate
+ *        from system boot)
+ *
+ * @param[in] aResp - The redfish response to return to the caller.
+ * @param[in] resourceName - The redfish resource name which trying to isolate.
+ * @param[in] resourceId - The redfish resource id which trying to isolate.
+ * @param[in] enabled - The redfish resource "Enabled" property value.
+ * @param[in] interfaces - The redfish resource dbus interfaces which will use
+ *                         to get the given resource dbus objects from
+ *                         the inventory.
+ *
+ * @return The redfish response in given response buffer.
+ *
+ * @note - This function will identify the given resource dbus object from
+ *         the inventory by using the given resource dbus interfaces along
+ *         with "Object:Enable" interface (which is used to map the "Enabled"
+ *         redfish property to dbus "Enabled" property - The "Enabled" is
+ *         used to do isolate the resource from system boot) and the given
+ *         redfish resource "Id".
+ *       - This function will do either isolate or deisolate based on the
+ *         given "Enabled" property value.
+ */
+inline void
+    processHardwareIsolationReq(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                                const std::string& resourceName,
+                                const std::string& resourceId, bool enabled,
+                                const std::vector<const char*>& interfaces)
+{
+    std::vector<const char*> resourceIfaces(interfaces.begin(),
+                                            interfaces.end());
+    resourceIfaces.push_back("xyz.openbmc_project.Object.Enable");
+
+    // Make sure the given resourceId is present in inventory
+    crow::connections::systemBus->async_method_call(
+        [aResp, resourceName, resourceId,
+         enabled](boost::system::error_code ec,
+                  const std::vector<std::string>& objects) {
+            if (ec || objects.empty())
+            {
+                BMCWEB_LOG_ERROR << "DBus response error [" << ec.value()
+                                 << " : " << ec.message()
+                                 << "] when tried to check the given resource "
+                                    "is present in tne inventory";
+                messages::internalError(aResp->res);
+                return;
+            }
+
```

Comment:
Nitpick: Shouldn't objects.empty() be a resource not found? 
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/44175/9/redfish-core/lib/systems.hpp#1989

Reply:
Actually, I should remove this check (checking in LN272) - will remove `objects.empty()`